cargo/src/cargo_commands.rs

```diff
@@ -0,0 +1,342 @@
+use std::fs;
+use std::io;
+use std::path::Path;
+use std::path::PathBuf;
+
+use sha3::Digest;
+use sha3::Keccak256;
+
+#[allow(unused_imports)]
+use tracing::{debug, error, info, trace, warn, Level};
+
+use crate::audit;
+use crate::vendor::has_dependencies;
+use crate::vendor::is_workspace;
+
+fn cargo_command(
+    subcommand: &str,
+    options: &[String],
+    curdir: impl AsRef<Path>,
+) -> io::Result<String> {
+    let cmd = std::process::Command::new("cargo")
+        .arg(subcommand)
+        .args(options.iter())
+        .current_dir(curdir.as_ref())
+        .output()?;
+    trace!(?cmd);
+    let stdoutput = String::from_utf8_lossy(&cmd.stdout);
+    let stderrput = String::from_utf8_lossy(&cmd.stderr);
+    if !cmd.status.success() {
+        debug!(?stdoutput);
+        debug!(?stderrput);
+        return Err(io::Error::new(io::ErrorKind::Interrupted, stderrput));
+    };
+    debug!(?stdoutput);
+    debug!(?stderrput);
+    // Return the output on success as this has the infor for .cargo/config
+    Ok(stdoutput.to_string())
+}
+
+pub fn cargo_fetch(curdir: &Path, manifest: &str, mut update: bool) -> io::Result<String> {
+    info!("⤵️ Running `cargo fetch`...");
+    let mut default_options: Vec<String> = vec![];
+    let manifest_path = PathBuf::from(&manifest);
+    let manifest_path_parent = manifest_path.parent().unwrap_or(curdir);
+    let possible_lockfile = manifest_path_parent.join("Cargo.lock");
+    if !update {
+        if possible_lockfile.is_file() {
+            default_options.push("--locked".to_string());
+        } else {
+            warn!(
+                                "⚠️ No lockfile present. This might UPDATE your dependencies. Overriding `update` from \
```

Comment:
Spacing seems weird, are you mixing tabs/spaces?

Reply:
![image](https://github.com/user-attachments/assets/142e4a07-aea2-41a3-8ae1-4ccbb7e90c7e)
they are spaces but i guess they are a left over from the other repo since i migrated the code there to here.

Reply:
cargo fmt?